public/components/trace_analytics/components/traces/trace_view.tsx

```diff
@@ -204,6 +206,22 @@ export function TraceView(props: TraceViewProps) {
     );
   };
 
+  const handlePayloadResponse = (data: string) => {
+    setPayloadData(data);
+    try {
+      const parsed = JSON.parse(data);
+      if (!Array.isArray(parsed) || parsed.length === 0) {
+        setTraceIdError(true);
+        setFields({});
+      } else {
+        setTraceIdError(false);
```

Comment:
Do we need the else part here? as the page starts with traceIdError: false

Reply:
handlePayloadResponse is part of the refresh so I think it is safer to maintain as the payload could be re-triggered if the mode changed.